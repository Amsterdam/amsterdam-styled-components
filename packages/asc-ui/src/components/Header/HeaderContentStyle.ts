import styled from '@datapunt/asc-core'
import { breakpoint } from '../../utils'

const HeaderContentStyle = styled.div`
  height: 50px;
  margin: 0;
  padding: 15px 25px 0 0;
  font-weight: 700;
  width: 70%;
  display: flex;
  justify-content: normal;
  align-self: flex-start;
  justify-content: flex-end;
  align-items: center;
  flex-shrink: 0;
  box-sizing: border-box;

  @media screen and ${breakpoint('max-width', 'laptopM')} {
    display: none;
  }
`
export default HeaderContentStyle
