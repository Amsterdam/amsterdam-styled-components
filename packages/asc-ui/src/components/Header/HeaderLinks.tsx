import React, { FunctionComponent, HTMLAttributes } from 'react'
import HeaderLinksStyle from './HeaderLinksStyle'

const HeaderLinks: FunctionComponent<HTMLAttributes<HTMLDivElement>> = ({
  children,
  ...otherProps
}) => <HeaderLinksStyle {...otherProps}>{children}</HeaderLinksStyle>

export default HeaderLinks
