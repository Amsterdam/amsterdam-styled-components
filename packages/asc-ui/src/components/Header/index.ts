import HeaderStyle from './HeaderStyle'
import HeaderLinksStyle from './HeaderLinksStyle'
import HeaderTitleStyle from './HeaderTitleStyle'
import HeaderLogoTextStyle from './HeaderLogoTextStyle'
import HeaderNavigationStyle from './HeaderNavigationStyle'

export { default } from './Header'

export const HeaderStyles = {
  HeaderLinksStyle,
  HeaderLogoTextStyle,
  HeaderNavigationStyle,
  HeaderTitleStyle,
  HeaderStyle,
}
