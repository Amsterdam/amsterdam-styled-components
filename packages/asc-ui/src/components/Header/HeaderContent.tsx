import { FunctionComponent, HTMLAttributes, ReactNode } from 'react'
import HeaderContentStyle from './HeaderContentStyle'

type HeaderContentProps = {
  children: ReactNode
} & HTMLAttributes<HTMLDivElement>

const HeaderContent: FunctionComponent<HeaderContentProps> = ({
  children,
  ...otherProps
}) => <HeaderContentStyle {...otherProps}>{children}</HeaderContentStyle>

export default HeaderContent
