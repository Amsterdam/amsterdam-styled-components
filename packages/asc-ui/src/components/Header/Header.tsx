import React from 'react'
import HeaderStyle, { Props as HeaderStyleProps } from './HeaderStyle'
import HeaderWrapperStyle, {
  Props as HeaderWrapperProps,
} from './HeaderWrapperStyle'
import HeaderLogoText, { LogoProps } from './HeaderLogoText'
import HeaderNavigation from './HeaderNavigation'
import HeaderLinks from './HeaderLinks'
import { CustomCssPropsType } from '../../utils'
import AmsterdamLogo from '../AmsterdamLogo'

type Props = {
  tall?: boolean
  homeLink: string
  title?: string
  navigation?: React.ReactNode
  links?: React.ReactNode
  logo?: React.FC<LogoProps>
} & HeaderWrapperProps &
  HeaderStyleProps &
  CustomCssPropsType &
  React.HTMLAttributes<HTMLDivElement>

const Header: React.FC<Props> = ({
  css,
  title,
  homeLink,
  fullWidth,
  tall,
  navigation,
  links,
  logo,
  ...otherProps
}) => (
  <HeaderWrapperStyle {...{ css, tall }} id="header">
    <HeaderStyle {...{ fullWidth, ...otherProps }}>
      <HeaderLogoText {...{ tall, title, homeLink, logo }} />
      <HeaderNavigation>{navigation}</HeaderNavigation>
      <HeaderLinks>{links}</HeaderLinks>
    </HeaderStyle>
  </HeaderWrapperStyle>
)

Header.defaultProps = {
  tall: false,
  fullWidth: true,
  logo: AmsterdamLogo,
}

export default Header
