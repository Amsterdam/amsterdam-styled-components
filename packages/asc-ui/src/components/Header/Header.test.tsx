import { render } from '@testing-library/react'
import Header from './Header'
import Typography from '../Typography'
import ThemeProvider from '../../theme/ThemeProvider'

const setup = (props = {}) =>
  render(
    <ThemeProvider>
      <Header
        title="Data en informatie"
        homeLink="http://data.amsterdam.nl"
        data-testid="header"
        {...props}
      >
        <Typography>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Amet
          debitis, dicta incidunt maxime necessitatibus voluptatibus.
        </Typography>
      </Header>
    </ThemeProvider>,
  )

describe('Header', () => {
  it('should render the short version', () => {
    const { getByTestId } = setup()

    expect(getByTestId('header')).toHaveStyleRule('max-width', 'none')
  })

  it('should render the short version without the full width', () => {
    const { getByTestId } = setup({ fullWidth: false })
    expect(getByTestId('header')).toHaveStyleRule('max-width', '1430px')
  })

  it('should render the tall version with a custom logo', () => {
    const CustomLogo = () => (
      <a href="/" data-testid="logo">
        some icon
      </a>
    )

    const { getByTestId } = setup({ tall: true, logo: CustomLogo })
    expect(getByTestId('logo')).toBeDefined()
  })
})
