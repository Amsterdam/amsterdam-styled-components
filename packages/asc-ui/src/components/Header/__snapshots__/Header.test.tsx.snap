// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the short version 1`] = `
.c10 {
  margin: 0;
  color: #000000;
  font-weight: inherit;
  font-size: 1rem;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.25;
  margin-bottom: 0;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 0;
  padding: 0 10px;
  background-color: #ffffff;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  height: 100%;
  margin: 0 auto;
  padding: 0 10px;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  max-width: none;
  padding-left: 10px !important;
}

.c2::after {
  content: '';
  display: block;
  min-height: inherit;
  font-size: 0;
}

.c2 .c14 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-bottom: 0;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
}

.c9:focus {
  background-color: #FEC813;
}

.c9 .c15 {
  margin: 3px;
}

.c9:hover {
  color: #ec0000;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c9:hover .c15 svg rect,
.c9:hover .c15 svg polygon,
.c9:hover .c15 svg path {
  fill: #ec0000;
}

.c8 {
  margin: 0px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8:hover {
  color: inherit;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  height: 100%;
}

.c6 {
  display: block;
  height: 100%;
  width: 100%;
  background-size: cover;
  background-image: url("");
}

.c5 {
  display: inline-block;
  height: 30px;
  width: 68px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c5:focus {
  z-index: 10;
  outline-color: #FEC813;
  outline-style: solid;
  outline-offset: 0px;
  outline-width: 3px;
}

.c3 {
  margin: 0;
  font-weight: 700;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: normal;
  -webkit-justify-content: normal;
  -ms-flex-pack: normal;
  justify-content: normal;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: content-box;
}

.c3 .c4 {
  margin-right: 10px;
}

.c0 {
  background-color: #ffffff;
  margin: 0 auto;
  width: 100%;
  max-width: none;
  box-shadow: 0 4px rgba(0,0,0,0.04);
  box-sizing: content-box;
  position: fixed;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  z-index: 2;
}

.c0 .c11 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-left: 15px;
}

.c0 .c7 {
  margin-top: -2px;
}

.c0 .c1 {
  min-height: 50px;
  padding-right: 0;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  top: 17px;
  right: 45px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c13 .c17 {
  height: 17px;
  padding: 0;
}

.c13 .c17:not(:last-child) {
  margin-right: 30px;
}

.c13 .c17:focus,
.c13 .c17:hover {
  background-color: transparent;
}

.c13 .c17:focus .c16,
.c13 .c17:hover .c16 {
  color: #ec0000;
}

.c13 .c17:focus svg rect,
.c13 .c17:hover svg rect,
.c13 .c17:focus svg polygon,
.c13 .c17:hover svg polygon,
.c13 .c17:focus svg path,
.c13 .c17:hover svg path {
  fill: #ec0000;
}

.c13 .c17 .c16 {
  line-height: 17px;
}

@media screen and (min-width:540.02px) {
  .c2 {
    padding: 0 15px;
  }
}

@media screen and (max-width:375px) {
  .c8 {
    font-size: 14px;
  }
}

@media screen and (min-width:375.02px) {
  .c8 {
    font-size: 16px;
  }
}

@media screen and (max-width:1200px) {
  .c13 {
    display: none;
  }
}

<div
  class="c0"
  id="header"
>
  <header
    class="c1 c2"
  >
    <h1
      class="c3"
    >
      <a
        class="c4 c5"
        href="http://data.amsterdam.nl"
      >
        <span
          class="c6"
        />
      </a>
      <a
        class="c7 c8 c9 c10"
        href="http://data.amsterdam.nl"
      >
        Data en informatie
      </a>
    </h1>
    <nav
      class="c11 c12"
    />
    <div
      class="c13"
    />
  </header>
</div>
`;

exports[`Header should render the short version without the full width 1`] = `
.c10 {
  margin: 0;
  color: #000000;
  font-weight: inherit;
  font-size: 1rem;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.25;
  margin-bottom: 0;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 0;
  padding: 0 10px;
  background-color: #ffffff;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  height: 100%;
  margin: 0 auto;
  padding: 0 10px;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  max-width: 1600px;
}

.c2::after {
  content: '';
  display: block;
  min-height: inherit;
  font-size: 0;
}

.c2 .c15 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-bottom: 0;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
}

.c9:focus {
  background-color: #FEC813;
}

.c9 .c16 {
  margin: 3px;
}

.c9:hover {
  color: #ec0000;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c9:hover .c16 svg rect,
.c9:hover .c16 svg polygon,
.c9:hover .c16 svg path {
  fill: #ec0000;
}

.c8 {
  margin: 0px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8:hover {
  color: inherit;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  height: 100%;
}

.c6 {
  display: block;
  height: 100%;
  width: 100%;
  background-size: cover;
  background-image: url("");
}

.c5 {
  display: inline-block;
  height: 30px;
  width: 68px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c5:focus {
  z-index: 10;
  outline-color: #FEC813;
  outline-style: solid;
  outline-offset: 0px;
  outline-width: 3px;
}

.c3 {
  margin: 0;
  font-weight: 700;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: normal;
  -webkit-justify-content: normal;
  -ms-flex-pack: normal;
  justify-content: normal;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: content-box;
}

.c3 .c4 {
  margin-right: 10px;
}

.c14 .c11 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-left: 15px;
}

.c14 .c7 {
  margin-top: -2px;
}

.c14 .c1 {
  min-height: 50px;
  padding-right: 0;
}

.c0 {
  background-color: #ffffff;
  margin: 0 auto;
  width: 100%;
  max-width: 1800px;
  box-shadow: 0 4px rgba(0,0,0,0.04);
  box-sizing: content-box;
  position: fixed;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  z-index: 2;
}

.c0 .c11 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-left: 15px;
}

.c0 .c7 {
  margin-top: -2px;
}

.c0 .c1 {
  min-height: 50px;
  padding-right: 0;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  top: 17px;
  right: 45px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c13 .c18 {
  height: 17px;
  padding: 0;
}

.c13 .c18:not(:last-child) {
  margin-right: 30px;
}

.c13 .c18:focus,
.c13 .c18:hover {
  background-color: transparent;
}

.c13 .c18:focus .c17,
.c13 .c18:hover .c17 {
  color: #ec0000;
}

.c13 .c18:focus svg rect,
.c13 .c18:hover svg rect,
.c13 .c18:focus svg polygon,
.c13 .c18:hover svg polygon,
.c13 .c18:focus svg path,
.c13 .c18:hover svg path {
  fill: #ec0000;
}

.c13 .c18 .c17 {
  line-height: 17px;
}

@media screen and (min-width:540.02px) {
  .c2 {
    padding: 0 15px;
  }
}

@media screen and (max-width:375px) {
  .c8 {
    font-size: 14px;
  }
}

@media screen and (min-width:375.02px) {
  .c8 {
    font-size: 16px;
  }
}

@media screen and (max-width:1200px) {
  .c13 {
    display: none;
  }
}

<div
  class="c0"
  id="header"
>
  <header
    class="c1 c2"
  >
    <h1
      class="c3"
    >
      <a
        class="c4 c5"
        href="http://data.amsterdam.nl"
      >
        <span
          class="c6"
        />
      </a>
      <a
        class="c7 c8 c9 c10"
        href="http://data.amsterdam.nl"
      >
        Data en informatie
      </a>
    </h1>
    <nav
      class="c11 c12"
    />
    <div
      class="c13"
    />
  </header>
</div>
`;

exports[`Header should render the tall version 1`] = `
.c13 {
  margin: 0;
  color: #000000;
  font-weight: inherit;
  font-size: 1rem;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.25;
  margin-bottom: 0;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 0;
  padding: 0 10px;
  background-color: #ffffff;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  height: 100%;
  margin: 0 auto;
  padding: 0 10px;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  max-width: none;
  padding-left: 10px !important;
}

.c3::after {
  content: '';
  display: block;
  min-height: inherit;
  font-size: 0;
}

.c3 .c20 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-bottom: 0;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
}

.c12:focus {
  background-color: #FEC813;
}

.c12 .c21 {
  margin: 3px;
}

.c12:hover {
  color: #ec0000;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c12:hover .c21 svg rect,
.c12:hover .c21 svg polygon,
.c12:hover .c21 svg path {
  fill: #ec0000;
}

.c11 {
  margin: 0px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c11:hover {
  color: inherit;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  height: 100%;
}

.c9 {
  display: block;
  height: 100%;
  width: 100%;
  background-size: cover;
  background-image: url("");
}

.c7 {
  display: inline-block;
  height: 30px;
  width: 68px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c7:focus {
  z-index: 10;
  outline-color: #FEC813;
  outline-style: solid;
  outline-offset: 0px;
  outline-width: 3px;
}

.c17 .c6 {
  margin-right: 10px;
}

.c5 {
  margin: 0;
  font-weight: 700;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: normal;
  -webkit-justify-content: normal;
  -ms-flex-pack: normal;
  justify-content: normal;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: content-box;
}

.c5 .c6 {
  margin-right: 10px;
}

.c18 .c14 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-left: 15px;
}

.c18 .c10 {
  margin-top: -2px;
}

.c18 .c2 {
  min-height: 50px;
  padding-right: 0;
}

.c19 .c14 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-left: 15px;
}

.c19 .c10 {
  margin-top: -2px;
}

.c19 .c2 {
  min-height: 50px;
  padding-right: 0;
}

.c0 {
  background-color: #ffffff;
  margin: 0 auto;
  width: 100%;
  max-width: none;
  box-shadow: 0 4px rgba(0,0,0,0.04);
  box-sizing: content-box;
  position: fixed;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  z-index: 2;
}

.c0 .c14 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-left: 15px;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  top: 17px;
  right: 45px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c16 .c23 {
  height: 17px;
  padding: 0;
}

.c16 .c23:not(:last-child) {
  margin-right: 30px;
}

.c16 .c23:focus,
.c16 .c23:hover {
  background-color: transparent;
}

.c16 .c23:focus .c22,
.c16 .c23:hover .c22 {
  color: #ec0000;
}

.c16 .c23:focus svg rect,
.c16 .c23:hover svg rect,
.c16 .c23:focus svg polygon,
.c16 .c23:hover svg polygon,
.c16 .c23:focus svg path,
.c16 .c23:hover svg path {
  fill: #ec0000;
}

.c16 .c23 .c22 {
  line-height: 17px;
}

@media screen and (min-width:540.02px) {
  .c3 {
    padding: 0 15px;
  }
}

@media screen and (max-width:375px) {
  .c11 {
    font-size: 14px;
  }
}

@media screen and (min-width:375.02px) {
  .c11 {
    font-size: 16px;
  }
}

@media screen and (min-width:1200.02px) {
  .c7 {
    height: 68px;
    width: 100px;
  }

  .c7 .c8 {
    background-image: url("");
  }
}

@media screen and (min-width:1200.02px) {
  .c5 .c6 {
    margin-right: 40px;
  }
}

@media screen and (max-width:1200px) {
  .c0 .c10 {
    margin-top: -2px;
  }

  .c0 .c2 {
    min-height: 50px;
    padding-right: 0;
  }
}

@media screen and (min-width:1200.02px) {
  .c0 {
    box-shadow: none;
  }

  .c0:after {
    content: '';
    display: block;
    position: absolute;
    left: 0;
    right: 0;
    height: 50px;
    margin-top: -50px;
    background-color: #F5F5F5;
  }

  .c0 .c2 {
    z-index: 100;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
  }

  .c0 .c1 {
    background-color: transparent;
  }

  .c0 .c10 {
    margin-top: 0px;
    -webkit-align-self: flex-start;
    -ms-flex-item-align: start;
    align-self: flex-start;
    line-height: 38px;
    font-size: 30px;
  }

  .c0 .c4 {
    height: 78px;
    padding: 15px 0;
  }

  .c0 .c14 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    margin-left: 25px;
    margin-right: 25px;
    height: 50px;
  }
}

@media screen and (max-width:1200px) {
  .c16 {
    display: none;
  }
}

<div
  class="c0"
  id="header"
>
  <header
    class="c1 c2 c3"
  >
    <h1
      class="c4 c5"
    >
      <a
        class="c6 c7"
        href="http://data.amsterdam.nl"
      >
        <span
          class="c8 c9"
        />
      </a>
      <a
        class="c10 c11 c12 c13"
        href="http://data.amsterdam.nl"
      >
        Data en informatie
      </a>
    </h1>
    <nav
      class="c14 c15"
    />
    <div
      class="c16"
    />
  </header>
</div>
`;

exports[`Header should render the tall version without a title 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 0;
  padding: 0 10px;
  background-color: #ffffff;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  height: 100%;
  margin: 0 auto;
  padding: 0 10px;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  max-width: none;
  padding-left: 10px !important;
}

.c3::after {
  content: '';
  display: block;
  min-height: inherit;
  font-size: 0;
}

.c3 .c16 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-bottom: 0;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  height: 100%;
}

.c9 {
  display: block;
  height: 100%;
  width: 100%;
  background-size: cover;
  background-image: url("");
}

.c7 {
  display: inline-block;
  height: 30px;
  width: 68px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c7:focus {
  z-index: 10;
  outline-color: #FEC813;
  outline-style: solid;
  outline-offset: 0px;
  outline-width: 3px;
}

.c13 .c6 {
  margin-right: 10px;
}

.c5 {
  margin: 0;
  font-weight: 700;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: normal;
  -webkit-justify-content: normal;
  -ms-flex-pack: normal;
  justify-content: normal;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: content-box;
}

.c5 .c6 {
  margin-right: 10px;
}

.c14 .c10 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-left: 15px;
}

.c14 .c2 {
  min-height: 50px;
  padding-right: 0;
}

.c15 .c10 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-left: 15px;
}

.c15 .c2 {
  min-height: 50px;
  padding-right: 0;
}

.c0 {
  background-color: #ffffff;
  margin: 0 auto;
  width: 100%;
  max-width: none;
  box-shadow: 0 4px rgba(0,0,0,0.04);
  box-sizing: content-box;
  position: fixed;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  z-index: 2;
}

.c0 .c10 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-left: 15px;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  top: 17px;
  right: 45px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c12 .c18 {
  height: 17px;
  padding: 0;
}

.c12 .c18:not(:last-child) {
  margin-right: 30px;
}

.c12 .c18:focus,
.c12 .c18:hover {
  background-color: transparent;
}

.c12 .c18:focus .c17,
.c12 .c18:hover .c17 {
  color: #ec0000;
}

.c12 .c18:focus svg rect,
.c12 .c18:hover svg rect,
.c12 .c18:focus svg polygon,
.c12 .c18:hover svg polygon,
.c12 .c18:focus svg path,
.c12 .c18:hover svg path {
  fill: #ec0000;
}

.c12 .c18 .c17 {
  line-height: 17px;
}

@media screen and (min-width:540.02px) {
  .c3 {
    padding: 0 15px;
  }
}

@media screen and (min-width:1200.02px) {
  .c7 {
    height: 68px;
    width: 100px;
  }

  .c7 .c8 {
    background-image: url("");
  }
}

@media screen and (min-width:1200.02px) {
  .c5 .c6 {
    margin-right: 40px;
  }
}

@media screen and (max-width:1200px) {
  .c0 .sc-bZQynM {
    margin-top: -2px;
  }

  .c0 .c2 {
    min-height: 50px;
    padding-right: 0;
  }
}

@media screen and (min-width:1200.02px) {
  .c0 {
    box-shadow: none;
  }

  .c0:after {
    content: '';
    display: block;
    position: absolute;
    left: 0;
    right: 0;
    height: 50px;
    margin-top: -50px;
    background-color: #F5F5F5;
  }

  .c0 .c2 {
    z-index: 100;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
  }

  .c0 .c1 {
    background-color: transparent;
  }

  .c0 .sc-bZQynM {
    margin-top: 0px;
    -webkit-align-self: flex-start;
    -ms-flex-item-align: start;
    align-self: flex-start;
    line-height: 38px;
    font-size: 30px;
  }

  .c0 .c4 {
    height: 78px;
    padding: 15px 0;
  }

  .c0 .c10 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    margin-left: 25px;
    margin-right: 25px;
    height: 50px;
  }
}

@media screen and (max-width:1200px) {
  .c12 {
    display: none;
  }
}

<div
  class="c0"
  id="header"
>
  <header
    class="c1 c2 c3"
  >
    <h1
      class="c4 c5"
    >
      <a
        class="c6 c7"
        href="http://data.amsterdam.nl"
      >
        <span
          class="c8 c9"
        />
      </a>
    </h1>
    <nav
      class="c10 c11"
    />
    <div
      class="c12"
    />
  </header>
</div>
`;
